planb/main.m

```diff
@@ -55,42 +99,28 @@ int main(int argc, const char * argv[]) {
     authURLSession.serverRootsPemData = [NSData dataWithBytes:ROOTS_PEM length:ROOTS_PEM_len];
     authURLSession.refusesRedirects = YES;
     authURLSession.loggingBlock = ^(NSString *line) {
-      PBLog(@"%@", line);
+      PBLog(@"Session: %@", line);
     };
 
-    NSArray *packages = @[
-      @[ @"pkg1/package1", @"com.megacorp.package1" ],
-      @[ @"pkg2/package2", @"com.megacorp.package2" ],
-      @[ @"pkg3/package3", @"com.megacorp.package3" ],
-    ];
-
-    for (NSArray *packageTuple in packages) {
-      NSString *item = [packageTuple objectAtIndex:0];
-      NSString *receiptName = [packageTuple objectAtIndex:1];
-      NSURL *url = [URLBuilder URLForTrackWithPkg:item];
-
-      PBLog(@"Requesting %@", url.absoluteString);
-
-      [[authURLSession.session downloadTaskWithURL:url completionHandler:^(NSURL *location,
-                                                                           NSURLResponse *response,
-                                                                           NSError *error) {
-        PBPackageInstaller *pkg = [[PBPackageInstaller alloc] initWithReceiptName:receiptName
-                                                                      packagePath:location.path
-                                                                     targetVolume:@"/"];
-        [pkg installApplication];
-        installComplete = YES;
-
-        PBLog(@"Finished with %@", item);
-      }] resume];
-
-      installComplete = NO;
-
-      while (!installComplete) {
-        [[NSRunLoop mainRunLoop] runUntilDate:[NSDate dateWithTimeIntervalSinceNow:0.5]];
-      }
-    }
-
-    return 0;
+    __block BOOL success = YES;
+    NSArray *packages = Packages();
+    [packages enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop) {
```

Comment:
Doesn't seem to make any difference in memory use and dealloc seems to be called on the PBPackageInstaller objects when expected.